velox/common/memory/SharedArbitrator.cpp

```diff
@@ -374,7 +409,8 @@ int64_t SharedArbitrator::reclaimableFreeCapacity(
     const MemoryPool& pool,
     bool isSelfReclaim) const {
   return std::min<int64_t>(
-      pool.freeBytes(), maxReclaimableCapacity(pool, isSelfReclaim));
+      getCapacityShrinkTarget(pool, pool.capacity()),
```

Comment:
We probably need to pass selfReclaim to getCapacityShrinkTarget? Or at the end of memory arbitration before it fails, we check if we can satisfy the request by calling maybeGrowFromSelf before we fail the query? Thanks!